nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -107,6 +112,74 @@ def write_entry(profile, generation, machine_id):
             f.write("machine-id %s\n" % machine_id)
     os.rename(tmp_path, entry_file)
 
+def write_secureboot_entry(profile, generation, machine_id):
+    if profile:
+        entry_file = "@efiSysMountPoint@/loader/entries/nixos-%s-generation-%d.conf" % (profile, generation)
+        efi_file_relative = "EFI/nixos/nixos-%s-generation-%d.efi" % (profile, generation)
+        efi_file = "@efiSysMountPoint@/%s" % (efi_file_relative)
+    else:
+        entry_file = "@efiSysMountPoint@/loader/entries/nixos-generation-%d.conf" % (generation)
+        efi_file_relative = "EFI/nixos/nixos-generation-%d.efi" % (generation)
+        efi_file = "@efiSysMountPoint@/%s" % (efi_file_relative)
+
+    initrd = "%s.initrd.tmp" % (efi_file)
+    shutil.copyfile(
+        profile_path(profile, generation, "initrd"),
+        initrd
+    )
+
+    try:
+        append_initrd_secrets = profile_path(profile, generation, "append-initrd-secrets")
+        subprocess.check_call([append_initrd_secrets, "@efiSysMountPoint@%s" % (initrd)])
+    except FileNotFoundError:
```

Comment:
I would prefer an explicity check if the file exists:

```
if os.path.exists(append_initrd_secrets):
    subprocess.check_call([append_initrd_secrets, "@efiSysMountPoint@%s" % (initrd)])
```

Reply:
Done!